### ISSUE TITLE
Checks could be more accurate/concise for flow 53 componentType

### ISSUE THREAD

**Issue by rosskevin** [brigand/babel-plugin-flow-react-proptypes#129]
```javascript
import type { ComponentType } from 'react';

type Props = {
    component?: string | ComponentType<*>,
}
```

This means any of a string, Component, or Stateless functional component is ok.

It generates: 

```
var babelPluginFlowReactPropTypes_proptype_ComponentType = require('react').babelPluginFlowReactPropTypes_proptype_ComponentType || require('prop-types').any;

component: require('prop-types').oneOfType([require('prop-types').string, typeof babelPluginFlowReactPropTypes_proptype_ComponentType === 'function' ? babelPluginFlowReactPropTypes_proptype_ComponentType : require('prop-types').shape(babelPluginFlowReactPropTypes_proptype_ComponentType)]),
```

Which is effectively:

```
component: require('prop-types').oneOfType([require('prop-types').string, require('prop-types').any])
```

I think we could more accurately and concisely limit this to 

```javascript
component: require('prop-types').oneOfType([require('prop-types').string, require('prop-types').func])
```

Similarly, the same could be said for `Element`.

This isn't a show stopper but nice to have.

**Comment by brigand** [comment]
Good idea! Published as 5.1.0.

**Comment by brigand** [comment]
Oops. I just realized I forgot to check the imported identifier's name (ComponentType). 5.1.1.
